src/rgw/rgw_sync_module_custom.cc

```diff
@@ -0,0 +1,434 @@
+
+#include <algorithm>
+#include <cstddef>
+
+#include "rgw_common.h"
+#include "rgw_coroutine.h"
+#include "rgw_cr_rados.h"
+#include "rgw_sync_module.h"
+#include "rgw_data_sync.h"
+#include "rgw_sync_module_custom.h"
+#include "rgw_custom_access.h"
+#include "rgw_rest_client.h"
+#include "rgw_rest_conn.h"
+#include "rgw_custom_access.h"
+
+#include <boost/asio/yield.hpp>
+
+#define dout_context g_ceph_context
+#define dout_subsys ceph_subsys_rgw
+
+class RGWGetRemoteObjCB: public RGWGetDataCB {
+  bufferlist data;
+  uint64_t len;
+public:
+  RGWGetRemoteObjCB():len(0) {}
+  ~RGWGetRemoteObjCB() { data.clear(); }
+  
+  bufferlist& get_data() { return data; }
+  uint64_t get_len() { return len; } 
+  
+  int handle_data(bufferlist& _bl, off_t _ofs, off_t _len) {
+    bufferptr bp(_bl.c_str(), _len);
+    data.push_back(bp);
```

Comment:
@platoxie this wouldn't scale, you'd end up holding the entire object in memory. What you need to do is feed this data to the sender, and have the sender stream out the data.